nilearn/regions/region_extractor.py

```diff
@@ -0,0 +1,320 @@
+        if extract_type == 'local_regions':
+            smooth_map = _smooth_array(map_, affine, peak_local_smooth)
+            seeds = _peak_local_max(smooth_map, indices=False)
+            seeds_label, seeds_id = label(seeds)
+            # Assign -1 to values which are 0. to indicate to ignore
+            seeds_label[map_ == 0.] = -1
+            rw_maps = _random_walker(map_, seeds_label, mode='cg')
+            # Now simply replace "-1" with "0" for regions seperation
+            rw_maps[rw_maps == -1] = 0.
+            label_maps = rw_maps
+        else:
+            # Taking directly the input data which is the case in 'auto' type
+            maps_assigned = map_
+            # Connected component extraction
+            label_maps, n_labels = label(maps_assigned)
+
+        # Takes the size of each labelized region data
+        labels_size = np.bincount(label_maps.ravel())
+        labels_size[0] = 0.
+        for label_id, label_size in enumerate(labels_size):
+            if label_size > min_size:
+                region_data = (label_maps == label_id) * map_
+                region_img = new_img_like(maps_img, region_data)
+                regions.append(region_img)
+
+        index_of_each_map.extend([index] * len(regions))
+        all_regions_imgs.extend(regions)
+
+    regions_extracted = concat_niimgs(all_regions_imgs)
+
+    return regions_extracted, index_of_each_map
+
+
+class RegionExtractor(NiftiMapsMasker):
+    """ Class to extract connected regions into seperate regions.
+
+    Region Extraction is a post processing technique which
+    is implemented to automatically segment each brain atlas maps
+    into different set of seperated brain activated region.
+    Particularly, to show that each decomposed brain maps can be
+    used to focus on a target specific Regions of Interest analysis.
+
+    Parameters
+    ----------
+    maps_img: 4D Niimg-like object or path to the 4D Niimg
+       an image or a filename of the image which contains a set of whole brain
+       atlas maps or statistically decomposed brain maps.
+
+    mask_img: Niimg-like object or None,  default is None, optional
+        Mask to be applied on the input data.
+        If given, mask_img is submitted to an instance of NiftiMapsMasker
+        for a preparation towards input maps data.
+        If None, no mask will be applied on the data.
+
+    min_size: int, default size 50, optional
+        An integer which suppress the smallest spurious regions based upon
+        the size of voxels. For example, min_size=50 means that regions
+        which have more than cluster of 50 voxels are kept.
+
+    threshold: a float or a string or a number, default string "80%" optional
+        A value used to threshold the input maps.
+        If given as float, this value will be used directly to threshold
+        the maps image. This float case is used in thresholding_strategy='img_value'
+        and 'ratio_n_voxels'.
+        or
```

Comment:
Not sure what you want to do with "or" but it's fine just to go to a new line so remove it. Same thing applies to "or" in other places.